src/helpers/shared/filters.js

```diff
@@ -0,0 +1,47 @@
+export const getFiltersFromUrl = (history, keys, defaultValues = {}) => {
+  const searchParams = new URLSearchParams(history.location.search);
+
+  let filters = keys.reduce((acc, key) => {
+    const values = searchParams.getAll(key);
+    return {
+      ...acc,
+      [key]: values.length > 1 ? values : values[0],
+    };
+  }, {});
+
+  Object.keys(defaultValues).forEach((key) => {
+    const value = defaultValues[key];
+    filters = {
+      ...filters,
+      [key]: Array.isArray(filters[key])
+        ? [...new Set([...filters[key], ...(Array.isArray(value) ? value : [value])])]
+        : (value?.length > 0 && value) || filters[key],
+    };
+
+    if (Array.isArray(value)) {
+      value.forEach((item) => searchParams.getAll(key).includes(item) || searchParams.append(key, item));
+    } else {
+      searchParams.get(key) || (value && searchParams.set(key, value));
+    }
+  });
+
+  history.replace(`${history.location.pathname}?${searchParams.toString()}`, {});
```

Comment:
Maybe I've misunderstood the purpose of this function, but why are we mutating the history search params in a function called `getFiltersFromUrl`? Shouldn't we change the search params in the other one?

Also, youou can use replace object as argument instead of constructing the whole query. Like so:
```jsx
history.replace({
  pathname: history.location.pathname,
  search: searchParams.toString()
})
```

Reply:
@Hyperkid123 the purpose is to get filter values from the URL and merge them with the default values provided. So maybe renaming the function to eg. `syncDefaultFiltersWithUrl` would be enough. I think that splitting the function does not really make sense there.